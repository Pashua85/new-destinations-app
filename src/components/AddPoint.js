import React from 'react';

export default class AddPoint extends React.Component {
  state = {
    point: ''
  };

  handleFormSubmit = (event) => {
    event.preventDefault();
    if(this.state.point) {
      this.props.handleAddPoint(this.state.point);
      this.setState({ point: '' });
    } else {
      this.props.handleError('Please add a route point');
    }
  };

  handleChange = (e) => {
    this.setState({ point: e.target.value });
  };

  render() {
    return (
      <form 
        className="add-point-form"
        onSubmit={this.handleFormSubmit}
      >
          <input 
            type="text" 
            className="add-point__input" 
            value={this.state.point}
            onChange={this.handleChange} 
            placeholder="add a route point..." 
          />
          <button className="add-point__btn">Add</button>
      </form>
    )
  }
}