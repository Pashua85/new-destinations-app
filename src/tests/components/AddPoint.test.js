import React from 'react';
import { shallow } from 'enzyme';
import AddPoint from '../../components/AddPoint';

let wrapper, handleError, handleAddPoint;

beforeEach(() => {
  handleError = jest.fn();
  handleAddPoint = jest.fn();
  wrapper = shallow(
    <AddPoint handleError={handleError} handleAddPoint={handleAddPoint}/>
  );
});

test('should render AddPoint correctly', () => {
  expect(wrapper).toMatchSnapshot();
});

test('should call handleError for empty form submition', () => {
  wrapper.find('form').simulate('submit', {
    preventDefault: () => {}
  });
  expect(handleError).toHaveBeenLastCalledWith('Please add a route point');  
});

test('should set new state on input value change', () => {
  wrapper.find('input').simulate('change', {
    target: {
      value: 'Test point'
    }
  });
  expect(wrapper.state('point')).toBe('Test point');
});

test('should call handleAddPoint with new point name on submition', () => {
  wrapper.find('input').simulate('change', {
    target: {
      value: 'Test point'
    }
  });
  wrapper.find('form').simulate('submit', {
    preventDefault: () => {}
  });
  expect(handleAddPoint).toHaveBeenLastCalledWith('Test point');
});

