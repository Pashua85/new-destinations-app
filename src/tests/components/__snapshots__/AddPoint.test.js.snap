// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render AddPoint correctly 1`] = `
<form
  className="add-point-form"
  onSubmit={[Function]}
>
  <input
    className="add-point__input"
    onChange={[Function]}
    placeholder="add a route point..."
    type="text"
    value=""
  />
  <button
    className="add-point__btn"
  >
    Add
  </button>
</form>
`;
